controllers/common.go

```diff
@@ -312,14 +321,24 @@ func (r *SearchReconciler) createSecret(ctx context.Context, secret *corev1.Secr
 		Name:      secret.Name,
 		Namespace: secret.Namespace,
 	}, found)
-	if err != nil && errors.IsNotFound(err) {
-		err = r.Create(ctx, secret)
-		if err != nil {
-			log.Error(err, "Could not create secret")
-			return &reconcile.Result{}, err
+	if err != nil {
+		if errors.IsNotFound(err) {
+			err = r.Create(ctx, secret)
+			if err != nil {
+				log.Error(err, "Could not create secret")
+				return &reconcile.Result{}, err
+			}
+			log.V(2).Info("Created %s secret", secret.Name)
+			log.V(9).Info("Created secret %+v", secret)
```

Comment:
This will log the contents of the secret which could be a security problem!

Reply:
Even with V(9) ? @jlpadilla